src/presence/presenceModel.js

```diff
@@ -206,22 +208,23 @@ export function reducer(
     case RESET_ACCOUNT_DATA:
       return initialState;
 
-    case REGISTER_COMPLETE:
-      return (
-        action.data.presences
-        // TODO(#5102): Delete fallback once we enforce any threshold for
-        //   ancient servers we refuse to connect to. It was added in
-        //   #2878 (2018-11-16), but it wasn't clear even then, it seems,
-        //   whether any servers actually omit the data. The API doc
-        //   doesn't mention any servers that omit it, and our Flow types
-        //   mark it required.
-        || initialState
-      );
+    case REGISTER_COMPLETE: {
+      // TODO(#5102): Delete fallback once we enforce any threshold for
+      //   ancient servers we refuse to connect to. It was added in
+      //   #2878 (2018-11-16), but it wasn't clear even then, it seems,
+      //   whether any servers actually omit the data. The API doc
+      //   doesn't mention any servers that omit it, and our Flow types
+      //   mark it required.
+      const data = action.data.presences ?? {};
```

Comment:
Hmm, I think the TODO is ripe for taking care of: we do enforce a threshold for ancient servers we refuse to connect to. (I noticed this when I saw the change from `||` to `??` and wondered if that might change any behavior, given our imperfect knowledge of ancient servers.)

Reply:
Hmm yeah, I guess that's right. I'll add a commit for that.